In [5]:
import numpy as np
from math import factorial

В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4 мяча. Какова вероятность того, что 3 мяча белые?

Пусть 
A-искомое событие "из двух корзин вытащили 3 белых" мяча. Тогда это событие может произойти следующими способами:
а)A1 - из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча;
б)A2 из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча;
в)A3 из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч.
Вероятность события A будет определяться как сумма вероятностей этих событий.
Определим вероятность события A1:
$$P(A_1)=\frac{C^2_3}{C^2_8}\cdot\frac{C^3_5\cdot C^1_7}{C^4_{12}}$$

In [6]:
def combinations(n,k):
    return factorial(n)/(factorial(k)*factorial(n-k))

In [7]:
P1=(combinations(3,2)/combinations(8,2))*(combinations(5,3)*combinations(7,1)/combinations(12,4))
print(f'Вероятность события "из первой корзины не вытащили ни одного белого мяча'
      f', из второй вытащили 3 белых мяча"\nP(A1) = {P1: .4f}'
     )

Вероятность события "из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча"
P(A1) =  0.0152


Определим вероятность события A2:
$$P(A_2)=\frac{C^1_5\cdot C^1_3}{C^2_8}\cdot\frac{C^2_5\cdot C^2_7}{C^4_{12}}$$


In [9]:
P2=(combinations(5,1)*combinations(3,1)*combinations(5,2)*combinations(7,2))/(combinations(8,2)*combinations(12,4))
print(f'Вероятность события "из первой корзины вытащили 1 белый мяч'
      f', из второй вытащили 2 белых мяча"\nP(A2) = {P2: .4f}'
     )

Вероятность события "из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча"
P(A2) =  0.2273


Определим вероятность события A3:
$$P(A_3)=\frac{C^2_5}{C^2_8}\cdot\frac{C^1_5\cdot C^3_7}{C^4_{12}}$$

In [10]:
P3=(combinations(5,2)*combinations(5,1)*combinations(7,3))/(combinations(8,2)*combinations(12,4))
print(f'Вероятность события "из первой корзины вытащили 2 белых мяча'
      f', из второй вытащили 1 белый мяч"\nP(A3) = {P3: .4f}'
     )

Вероятность события "из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч"
P(A3) =  0.1263


Вычислим $$P(A)=P(A_1+A_2+A_3)=P(A_1)+P(A_2)+P(A_3)$$


In [11]:
print(f'Вероятность того, что из вытащенных мячей 3 белые Р(А) = {P1+P2+P3: .4f}')

Вероятность того, что из вытащенных мячей 3 белые Р(А) =  0.3687


3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен:
a) первым спортсменом;
б) вторым спортсменом;
в) третьим спортсменом.
Рассмотрим следующие события:
A-в результате выстрела, цель поражена;
B1 - выстрел произведен первым спортсменом;
B2- выстрел произведен вторым спортсменом;
B3 - выстрел произведен третьим спортсменом.
Вероятности того, что выстрел произведен определенным спортсменом равны, т.к. выстрел 1, и произвести его могут только 3 этих спортсмена, т.е. событияB1,B2,B3 составляют полную группу событий.
 $$\Rightarrow P(B_1)=P(B_2)=P(B_3)=\frac{1}{3}$$
Вероятность того, что произошло событие A, при условии наступления события Bn даны: $P_{B_1}(A)=0.9, P_{B_2}(A)=0.8, P_{B_3}(A)=0.6$
 и, по формуле полной вероятности, вероятность наступления события A $$ P(A)=P(B_1)\cdot P_{B_1}(A)+P(B_2)\cdot P_{B_2}(A)+P(B_3)\cdot P_{B_3}(A)=\frac{1}{3}\cdot 0.9+\frac{1}{3}\cdot 0.8+\frac{1}{3}\cdot 0.6$$

 


In [12]:
PB=1/3
PA=PB*0.9+PB*0.8+PB*0.6
print(f'Полная вероятность наступления события А Р(А) = {PA: .4f}')

Полная вероятность наступления события А Р(А) =  0.7667


Теперь, исходя из того, что событие А уже произошло, с помощью формулы Байеса оценим вероятность того, что имело место одно из событий В:
$$P_A(B_n)=\frac{P(B_n)\cdot P_{B_n}(A)}{P(A)}\Rightarrow\\P_A(B_1)=\frac{\frac{1}{3}\cdot 0.9}{0.7667}\\P_A(B_2)=\frac{\frac{1}{3}\cdot 0.8}{0.7667}\\P_A(B_1)=\frac{\frac{1}{3}\cdot 0.6}{0.7667}$$


In [13]:
PAB1=PB*0.9/PA
PAB2=PB*0.8/PA
PAB3=PB*0.6/PA
print(f'Вероятность того, что выстрел произвёл первый спортсмен: {PAB1: .4f};\n'
      f'Вероятность того, что выстрел произвёл второй спортсмен: {PAB2: .4f};\n'
      f'Вероятность того, что выстрел произвёл третий спортсмен: {PAB3: .4f}.'
     )

Вероятность того, что выстрел произвёл первый спортсмен:  0.3913;
Вероятность того, что выстрел произвёл второй спортсмен:  0.3478;
Вероятность того, что выстрел произвёл третий спортсмен:  0.2609.


4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится:
a) на факультете A;
б) на факультете B;
в) на факультете C?

Пусть 1/4 - количество студентов на факультете A, тогда на факультете B учится также 1/4 студентов, а на факультете С - 2/4 студентов.
 Полная вероятность наступления события (сессия сдана) исходя из формулы полной вероятности:
 $$P=0.25\cdot 0.8+0.25\cdot 0.7+0.5\cdot 0.9$$


In [17]:
P=0.25*0.8+0.25*0.7+0.5*0.9
print(f'Полная вероятность наступления события P = {P}.')

Полная вероятность наступления события P = 0.825.


По формуле Байеса вычислим вероятность того, на каком факультете учится студент, успешно сдавший сессию.

In [18]:
PDSA=0.25*0.8/P
PDSB=0.25*0.7/P
PDSC=0.5*0.9/P
print(f'Вероятность того, что студент учится на факультете А: {PDSA: .4f};\n'
      f'Вероятность того, что студент учится на факультете B: {PDSB: .4f};\n'
      f'Вероятность того, что студент учится на факультете C: {PDSC: .4f}.'
      )

Вероятность того, что студент учится на факультете А:  0.2424;
Вероятность того, что студент учится на факультете B:  0.2121;
Вероятность того, что студент учится на факультете C:  0.5455.


5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя:
а) все детали;
б) только две детали;
в) хотя бы одна деталь;
г) от одной до двух деталей?

Введем следующие события:
а) Событие A1-вышла из строя первая деталь, по условию задачи-вероятность этого события:0,1;
б) Событие $\bar A_1$ -противоположное событию A1вероятность этого события:1-0,1=0,9;
в) Событие A2-вышла из строя вторая деталь, по условию задачи-вероятность этого события: 0,2;
г) Событие $\bar A_2$-противоположное событию A2 - вторая деталь не вышла из строя, вероятность этого события: 1-0,2=0,8;
д) Событие A3 -вышла из строя третья деталь, по условию задачи-вероятность этого события: 0,25;
е) Событие $\bar A_3$ -противоположное событию  A3 - третья деталь не вышла из строя, вероятность этого события: 1-0,25=0,75

Для того, чтобы произошло событие "вышли из строя все детали", необходимо, чтобы одновременно произошли события А1,А2,А3. А вероятностью этого события будет произведением вероятностей этих событий.
$$P(3)=P(A_1)\cdot P(A_2)\cdot P(A_3)=0.1\cdot 0.2\cdot 0.25$$

In [19]:
P3=0.1*0.2*0.25
print(f'Вероятность того, что из строя выйдут все детали Р(3) = {P3: .4f}')

Вероятность того, что из строя выйдут все детали Р(3) =  0.0050


б) Для того, чтобы произошло событие "вышли из строя 2 детали", необходимо что бы случилось одна из следующих комбинаций событий:

Произошли события A1 ,A2, $\bar A_3$;
произошли события A1 ,A3,$\bar A_2$;
произошли события A2,A3,$\bar A_1$.
А вероятность этого события будет суммой вероятностей этих комбинаций событий.

$$P=P(A_1)\cdot P(A_2)\cdot P(\bar A_3)+P(A_1)\cdot P(A_3)\cdot P(\bar A_2)+P(A_2)\cdot P(A_3)\cdot P(\bar A_1)=0.1\cdot 0.2\cdot 0.75+0.1\cdot 0.25\cdot 0.8+0.2\cdot 0.25\cdot 0.9$$

In [20]:
P2=0.1*0.2*0.75+0.1*0.25*0.8+0.2*0.25*0.9
print(f'Вероятность того, что из строя выйдут  2 детали Р(2) = {P2: .4f}')

Вероятность того, что из строя выйдут  2 детали Р(2) =  0.0800


в) Событие "выйдет из строя хотя бы одна деталь" можно рассмотреть как противоположное событию "не вышло из строя ни одной детали".

 вероятность события "не вышло из строя ни одной детали":

$$P(0)=P(\bar A_1)\cdot P(\bar A_2)\cdot P(\bar A_3)=0.9*0.8*0.75$$

In [21]:
P0=0.9*0.8*0.75
print(f'Вероятность того, что из строя не выйдет ни одной детали Р(0) = {P0: .4f}')

Вероятность того, что из строя не выйдет ни одной детали Р(0) =  0.5400


Тогда вероятность события "выйдет из строя хотя бы одна деталь"

In [24]:
P=1-P0
print(f'Вероятность того, что выйдет из строя хотя бы одна деталь  = {P: .4f}')

Вероятность того, что выйдет из строя хотя бы одна деталь  =  0.4600


г) Событие "из строя выйдут от одной до двух деталей" произойдет при наступлении события "из строя выйдет одна деталь" или из строя выйдут 2 детали, следовательно его вероятность будет равна сумме вероятностей этих событий: 

$$P(1,2)=P(1)+P(2)$$

$P(2)=0.08$, найдем $P(1)$:

События "не выйдет из строя ни одной детали", "из строя выйдет одна деталь", "из строя выйдет 2 детали", "из строя выйдут все детали" - несовместны, и образуют полную группу событий, поэтому сумма вероятностей наступления этих событий равна 1.$P(0)+P(1)+P(2)+P(3)=1\\0.54+P(1)+0.08+0.005=1\\P(1)=0.375$

$P(1,2)=P(1)+P(2)=0.375+0.08=0.455$